stdlib/src/handlers/ecdsa.rs

```diff
@@ -0,0 +1,285 @@
+        ])
+    }
+}
+
+impl PrecompileVerifier for EcdsaPrecompile {
+    /// Verifier for ECDSA signature verification at verification time.
+    ///
+    /// Receives the serialized request data (public key || digest || signature) stored during
+    /// execution (see [`EventHandler::on_event`]), re-performs the ECDSA verification, and
+    /// generates a commitment `RPO(RPO(RPO(pk) || RPO(digest)) || RPO(sig))` with tag
+    /// `[event_id, result, 0, 0]` that validates against the execution trace. Each of `pk`,
+    /// `digest`, and `sig` is first converted to u32‑packed field elements before hashing.
+    fn verify(&self, calldata: &[u8]) -> Result<PrecompileCommitment, PrecompileError> {
+        let request = EcdsaRequest::read_from_bytes(calldata)?;
+        Ok(request.as_precompile_commitment())
+    }
+}
+
+/// ECDSA signature verification request containing all data needed to verify a signature.
+///
+/// This structure encapsulates a complete ECDSA verification request including the public key,
+/// message digest, and signature. It is used during both execution (via the event handler) and
+/// verification (via the precompile verifier).
+pub struct EcdsaRequest {
+    /// secp256k1 public key (33 bytes, compressed)
+    pk: PublicKey,
+    /// Message digest (32 bytes, typically Keccak256 hash)
+    digest: [u8; MESSAGE_DIGEST_LEN_BYTES],
+    /// ECDSA signature (serialized by the implementation; 66 bytes in this crate)
+    sig: Signature,
+}
+
+impl EcdsaRequest {
+    /// Creates a new ECDSA verification request.
+    ///
+    /// # Arguments
+    /// * `pk` - The secp256k1 public key (33 bytes, compressed)
+    /// * `digest` - The message digest (32 bytes)
+    /// * `sig` - The ECDSA signature
+    pub fn new(pk: PublicKey, digest: [u8; MESSAGE_DIGEST_LEN_BYTES], sig: Signature) -> Self {
+        Self { pk, digest, sig }
+    }
+
+    /// Returns a reference to the public key.
+    pub fn pk(&self) -> &PublicKey {
+        &self.pk
+    }
+
+    /// Returns a reference to the digest.
+    pub fn digest(&self) -> &[u8; MESSAGE_DIGEST_LEN_BYTES] {
+        &self.digest
+    }
+
+    /// Returns a reference to the signature.
+    pub fn sig(&self) -> &Signature {
+        &self.sig
+    }
+
+    /// Converts this request into a [`PrecompileRequest`] for deferred verification.
+    ///
+    /// Serializes the request data (public key || digest || signature) and wraps it in a
+    /// PrecompileRequest with the ECDSA event ID.
+    pub fn as_precompile_request(&self) -> PrecompileRequest {
+        let mut calldata = Vec::with_capacity(PRECOMPILE_REQUEST_LEN);
+        self.write_into(&mut calldata);
+        PrecompileRequest::new(ECDSA_VERIFY_EVENT_NAME.to_event_id(), calldata)
+    }
+
+    /// Performs ECDSA signature verification and returns the result.
+    ///
+    /// Returns `true` if the signature is valid for the given public key and digest,
+    /// `false` otherwise.
+    pub fn result(&self) -> bool {
```

Comment:
nit: we usually use non-action-verb method names for "properties" (i.e., simple accessors). Here, I'd probably use something like `compute_result()` or `perform_verification()` or something similar.